### PR TITLE
Fix completion streaming for agents and introduce quick response

### PR DESCRIPTION
**Pull request by dusvyat**
## Description

Refactor agent completion streaming logic for improved stability and error handling. Introduce quick responses and structured log messages, and ensure proper formatting of streamed chunks.

Corresponding changes introduced in SDK: https://github.com/mindsdb/mindsdb_python_sdk/pull/142

Part of ML-104

https://www.loom.com/share/eb8dd3eddc7643a592c88112725a93e4

## Type of change

(Please delete options that are not relevant)

- [x] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] ⚡ New feature (non-breaking change which adds functionality)
- [ ] 📢 Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] 📄 This change requires a documentation update

## Verification Process

To ensure the changes are working as expected:

 - [x]   Test Location: using retrieval agent example in python sdk
 - [ ]   Verification Steps: Outline the steps or queries needed to validate the change. Include any data, configurations, or actions required to reproduce or see the new functionality.

## Additional Media:

- [ ] I have attached a brief loom video or screenshots showcasing the new functionality or change.

## Checklist:

- [ ] My code follows the style guidelines(PEP 8) of MindsDB.
- [ ] I have appropriately commented on my code, especially in complex areas.
- [ ] Necessary documentation updates are either made or tracked in issues.
- [ ] Relevant unit and integration tests are updated or added.

### CODE REVIEW

File: mindsdb/integrations/utilities/rag/pipelines/rag.py [hunk 1]
```diff
@@ -9,8 +10,6 @@
 from mindsdb.integrations.utilities.rag.settings import RAGPipelineModel, DEFAULT_AUTO_META_PROMPT_TEMPLATE
 from mindsdb.integrations.utilities.rag.vector_store import VectorStoreOperator
 
-from mindsdb.interfaces.agents.safe_output_parser import SafeOutputParser
```
Comment (dusvyat) [inline mindsdb/integrations/utilities/rag/pipelines/rag.py:12]: This was the main issue behind the problem with streaming, streaming agent executor in langchain expects message to be a string not a dictionary.

File: mindsdb/interfaces/agents/langchain_agent.py [hunk 2]
```diff
@@ -439,33 +457,52 @@ def _invoke_agent_executor_with_prompt(agent_executor, prompt):
...
-        for col in input_variables:
-            kwargs[col] = first_row[col] if first_row[col] is not None else ''  # add empty quote if data is missing
-        prompt = prompt_template.format(**kwargs)
-        if not prompt:
-            return
-        for chunk in agent_executor.stream(prompt, config={'callbacks': self._get_agent_callbacks(args)}):
-            yield chunk
+        callbacks, context_callback = prepare_callbacks(self, args)
+
+        yield 'data: {}\n\n'.format(json.dumps({"type": "start", "prompt": prompts[0]}))
```
Comment (ea-rus) [inline mindsdb/interfaces/agents/langchain_agent.py:467]: a lot of json.dumps in this function. 
maybe let it return dict and do json.dumps in function who receives it (_completion_event_generator)

File: mindsdb/interfaces/agents/langchain_agent.py [hunk 3]
```diff
@@ -439,33 +457,52 @@ def _invoke_agent_executor_with_prompt(agent_executor, prompt):
...
+
+            # After the stream is complete, yield the captured context
+            captured_context = context_callback.get_contexts()
+            if captured_context:
+                yield 'data: {}\n\n'.format(json.dumps({
+                    "type": "context",
+                    "content": captured_context
+                }))
+
+        except AttributeError as e:
```
Comment (ea-rus) [inline mindsdb/interfaces/agents/langchain_agent.py:493]: _completion_event_generator already catches exceptions. we can just remove all 'except' blocks here and it should work the same. Error prefixes (for example "Stream method error:") could be moved to exception message

File: mindsdb/interfaces/agents/langchain_agent.py [hunk 4]
```diff
@@ -439,33 +457,52 @@ def _invoke_agent_executor_with_prompt(agent_executor, prompt):
...
+        except TypeError as e:
+            error_message = f"Stream iterator error: {str(e)}"
+            logger.error(error_message, exc_info=True)
+            yield 'data: {}\n\n'.format(json.dumps({"error": error_message}))
+
+        except Exception as e:
+            error_message = handle_agent_error(e)
+            yield 'data: {}\n\n'.format(json.dumps({"error": error_message}))
+
+        finally:
```
Comment (ea-rus) [inline mindsdb/interfaces/agents/langchain_agent.py:507]: if this message has to be sent: maybe it is safer to move it to _completion_event_generator or to route function.
potentially the stream_agent could not be called

File: mindsdb/interfaces/agents/langchain_agent.py [hunk 5]
```diff
@@ -439,33 +457,52 @@ def _invoke_agent_executor_with_prompt(agent_executor, prompt):
...
+
+            # Try to get the stream iterator
+            stream_iterator = agent_executor.stream(prompts[0], config={'callbacks': callbacks})
+
+            # Ensure it's iterable
+            if not hasattr(stream_iterator, '__iter__'):
+                raise TypeError("The stream method did not return an iterable")
+
+            for chunk in stream_iterator:
+                chunk_obj = process_chunk(chunk)
```
Comment (ea-rus) [inline mindsdb/interfaces/agents/langchain_agent.py:482]: what the process_chunk for? can the chunk contain generators?

**Pull request merged**